automaton.py

```diff
@@ -97,10 +115,160 @@ def validate(self) -> bool:
                 origin.add_transition(target.name, trans.symbol)
                 if trans.symbol == EPSILON_SYMBOL:
                     self._has_epsilon = True
+                    self.exec_path = NFA
+        for state in self.states:
+            if len(state.transitions) != len(self.alphabet):
```

Comment:
We should check that the len is the same and also that all of the alphabet symbols are covered.
Since we could have for the alphabet {0,1} two transitions to 0